vault.go

```diff
@@ -481,6 +481,44 @@ func newVaultClient(opts *config) (*api.Client, error) {
 	// step: set the token for the client
 	client.SetToken(token)
 
+	if opts.vaultRenewToken {
+		tokeninfo, err := client.Auth().Token().LookupSelf()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup token info: %s", err)
+		}
+
+		tokenttl, err := tokeninfo.TokenTTL()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup token ttl: %s", err)
+		}
+		glog.Infof("token ttl is %v", tokenttl)
+		renewPeriod := tokenttl / 2
+		go func() {
+			for {
+				glog.Infof("scheduling token renew in %v", renewPeriod)
+				<-time.After(renewPeriod)
+
+				glog.Infof("attempting token renew")
+				newtokeninfo, err := client.Auth().Token().RenewSelf(0)
```

Comment:
quick question ... looking at [RenewSelf](https://github.com/hashicorp/vault/blob/master/api/auth_token.go#L121-L136) on error it's a nil .. should there not be a ```continue``` in the [error catch](https://github.com/UKHomeOffice/vault-sidekick/pull/61/files#diff-f95859fde2c2af3fd61c1b714a161d65R503) other ```newtokeninfo.TokenTTL()``` is gonna cause a nil pointer and then chunk the ```if renewPeriod < 1*time.Second {}``` at the near top of the loop

Reply:
Yes, good ideas. Changes made.